dbt/scripts/export_qc_town_close_reports.py

```diff
@@ -203,13 +191,63 @@ def main():
             f"--json-string '{json.dumps(iasworld_deps)}' "
         )
     else:
-        where = f"township_code = '{args.township}' and taxyr = '{args.year}'"
-        export_models(
+        models_for_export = query_models_for_export(
             target=args.target,
             rebuild=args.rebuild,
-            select=[tag, f"{tag}_{tag_suffix}"],
-            where=where,
+            select=select,
+            where=f"taxyr = '{args.year}'",
         )
+        output_dir = args.output_dir or "export/output/"
+        for model in models_for_export:
+            for town in townships:
+                # Only produce an export for this town if the report model
+                # is configured for its reassessment year
+                town_is_tri = town.is_reassessed_during(args.year)
+                if f"{base_tag}_tri" in model.tags and not town_is_tri:
+                    print(
+                        f"Skipping {model.name} for {town.township_name} "
+                        "because the report is only for tri towns"
+                    )
+                    continue
+                elif f"{base_tag}_non_tri" in model.tags and town_is_tri:
+                    print(
+                        f"Skipping {model.name} for {town.township_name} "
+                        "because the report is only for non-tri towns"
+                    )
+                    continue
+
+                # Filter the query results for only this town, but only if the
+                # query results are not empty, since otherwise they will have
+                # no columns
+                town_df = (
+                    model.df
+                    if model.df.empty
+                    else model.df[
+                        model.df["township_code"] == town.township_code
+                    ]
+                )
```

Comment:
There were two ways I thought about filtering results by township:

1. As part of the Athena query that pulls results, as the previous iteration of this PR did
2. _After_ pulling data from Athena, as this code block does

Option 2 allows us to more easily export a model for each town, since filtering in Pandas allows us to avoid incurring the overhead of a bunch of extra Athena queries.